File: src/services/addintServer.cpp
#include <ros/ros.h>
#include <ros_basics/addTwoInts.h>

bool add(ros_basics::addTwoInts::Request &request, ros_basics::addTwoInts::Response &response){

    response.sum = request.a + request.b;
    ROS_INFO("request x=%d , y=%d", request.a, request.b);
    ROS_INFO("sending response = %d", response.sum);

    return true;

}

int main(int argc, char **argv){

    ros::init(argc, argv , "add_two_ints_server");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertise("add_two_ints",add);

    ROS_INFO("Ready to add two ints");
    ros::spin();
    
    return 0;
}

File: src/services/addintServer.py
#!/usr/bin/env python3

from ros_basics.srv import addTwoInts
from ros_basics.srv import addTwoIntsRequest
from ros_basics.srv import addTwoIntsResponse

import rospy
import time

def callbak_add_two_ints(req):
    
    print("Returning addition [%s + %s = %s]" %(req.a,req.b, (req.a+req.b)))
    
    time.sleep(5)
    return addTwoIntsResponse(req.a+req.b)

def intergerAdd():
    
    rospy.init_node('add_two_ints_server')
    
    service = rospy.Service('add_two_ints',addTwoInts,callbak_add_two_ints)
    print("ready to add ints")
    rospy.spin()
    
if __name__ == '__main__':
    intergerAdd()